docs/components/onclick.md

```diff
@@ -1,3 +1,28 @@
 # matestack core component: Action
 
-Show [specs](../../spec/usage/components/action_spec.rb)
+Show [specs](../../spec/usage/components/onclick_spec.rb)
+
+The `onclick` component renders a div that runs a function on click. This is a nice component for use cases where you would like to wrap multiple components into one click function. The function called by the onclick component requires two fields. These two fields are `emit` and `data`. These two fields are then emitted as a event in javascript when the component is clicked
```

Comment:
1. The `onclick` only requires `emit`. `data` is optional
2. The usecase you describe sounds pretty complicated. The main purpose of this component is to provide an easy to use client side event emitting API. Normally it would only wrap one component, which shows an interaction element such as one button or one specially styled div or something

```ruby
onclick emit: "my_event" do
    button text: "click"
end
async rerender_on: "my_event" do
    #some content which gets rerendered onclick
end
```
3. If you want to show the more advanced usage of `onclick` including `data` param usages and a separate method defining the config of the `onclick` component, please use a second example and increase the complexity of your examples step by step. Otherwise you may "overwhelm" new matestack users
4. please add a separate section where you explain which params could be used for the `onclick` component. We use this approach in other component documentations as well :)